paddle/operators/lstm_op.h

```diff
@@ -109,15 +115,23 @@ class LSTMKernel : public framework::OpKernel<T> {
 
       int cur_batch_size = bend - bstart;
 
-      if (n != 0) {
+      if (n > 0) {
         int pre_h_start = static_cast<int>(batch_starts[n - 1]);
         int pre_h_end = pre_h_start + cur_batch_size;
         auto pre_hidden_t = batch_hidden.Slice(pre_h_start, pre_h_end);
         math::matmul<Place, T>(device_ctx, pre_hidden_t, false, *weight, false,
                                static_cast<T>(1.0), &gate_t,
                                static_cast<T>(1.0));
+      } else if (hidden_t0) {
```

Comment:
Since `H0` is optional, what will happen if `hidden_t0` is nullptr?

Reply:
`hidden_t0` is nullptr, means there is no hidden initialization, that is to say the `hidden_t0`  is zero. And there is no need to compute `matmul` in line 122.